tutorial.md

```diff
@@ -0,0 +1,191 @@
+# Hello Cloud Demo Step-by-Step Tutorial
+
+This is a step-by-step tutorial on how to build and deploy this project to the cloud
+
+---
+
+# <a name="prereq"></a> Pre-Requisites
+
+- Docker
+- DockerHub Account. Click [here](https://hub.docker.com/) to create account
+- Java
+- Git
+- curl
+- Maven
+- helm. Install helm by following the instructions [here](https://helm.sh/docs/intro/install/)
+- Cloudant database
+- IBM Cloud Account. Click [here](https://cloud.ibm.com) to create account
+- IBM cloud CLI (IBM cloud command line interface)
+- kubectl (kubernetes command line interface)
+- Kubernetes Cluster
+
+---
+
+### If you already have the above pre-requisites, you can jump to [Step-by-Step](#stepbystep)
+
+---
+
+### 0. Make sure you created the accounts highlighted in [Pre-Requisites](#prereq), they are DockerHub and IBM Cloud Account, and install helm.
+
+### 1. Install IBM cloud CLI interface ibmcloud 
+```
+curl -fsSL https://clis.cloud.ibm.com/install/osx | sh
+```
+Note: This instructions assumes user has Mac OSX. To install IBM Cloud CLI in another platform please visit [platforms](https://cloud.ibm.com/docs/cli?topic=cli-install-ibmcloud-cli) and check your OS equivalent command line option to install ibmcloud CLI
+
+### 2. Install kubernetes service plugin (might need to run with sudo) kubectl
+```
+ibmcloud plugin install kubernetes-service
+```
+
+### 3. <a name="step3"></a>In a folder of your choosing Clone this project hello cloud
+```
+git clone https://github.com/cloudreadyapp/hello_cloud_demo.git
+cd hello_cloud_demo
+```
+
+### 4. Login into your IBM account
+```
+ibmcloud login
+```
+Note: If there’s an error message: “ou are using a federated user ID, please use one time passcode ( ibmcloud login --sso )” use
+```
+ibmcloud login --sso
+```
+
+### 5.0 Create a kubernetes cluster (You can also do it via the dashboard UI at https://cloud.ibm.com). NOTE: You can only have 1 Lite Kubernetes cluster running at a time. If you already have one already, use the existing one or delete it.
+```
+ibmcloud ks cluster create classic --name hello-cloud-app-cluster
+```
+Note: If creating cluster for the first time, it will take a while, ~20 minutes. Cluster is ready once it's in normal state.
+### 5.1 <a name="checkStatus"></a>To check the status of the cluster, type
+```
+ibmcloud ks clusters
+```
+
+### 6. Create a Cloudant Database. You can do so by following this link [here](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-using-the-ibm-cloudant-dashboard)
+
+Before continuing with the tutorial, make sure you have the appropriate `server.env` file under `database/src/main/liberty/config/server.env` with the appropriate database constants that you created in step 6. It should look something like
```

Comment:
Suggestd edit:

Now, create a file `server.env` in [`database/src/main/liberty/config/`](https://github.com/cloudreadyapp/hello_cloud_demo/tree/master/database/src/main/liberty/config). This file will hold your Cloudant database constants: API Key, Endpoint URL and Database Name.

---

Maybe we should create a blank file there so they just need to edit and input their Cloudant details?

Reply:
Probably not, because we would need to remove `server.env` from `.gitignore`, and they might mistakenly push `server.env` with their database api keys to github which would be an issue

Reply:
We don't have to remove it from the gitignore, we can force add the file and commit it. Then they can just add their details instead of having to create the file in the right place. It'll be in the gitignore so they won't be able to push it unless they force push it themselves

Reply:
Just a suggestion though, we can keep it in the tutorial.md file if that's preferred